thirdparty/datastore2/datastore_closer.go

```diff
@@ -33,3 +34,8 @@ func (w *datastoreCloserWrapper) Batch() (datastore.Batch, error) {
 
 	return bds.Batch()
 }
+
+// TODO(geoah) Move to ./testutils or somewhere else
+func ThreadSafeCloserMapDatastore() ThreadSafeDatastoreCloser {
+	return CloserWrap(syncds.MutexWrap(datastore.NewMapDatastore()))
+}
```

Comment:
This is the `ThreadSafeCloserMapDatastore` I'm talking about.
Originally this was under `thirdparty/testutils/datastore.go` but as there was no other reason for `go-testutils` to depend on `go-datastore` it was removed.

I have temporarily put it under `thirdparty/datastore2` but I'm open to suggestions.